lib/arrays.md

```diff
@@ -5,44 +5,48 @@ instructor_notes: Feel free to re-organize the headings (or add/remove headings)
 
 # What is an Array?
 
-Replace me with your response...
+An array is a collection of objects that are indexed by integer, starting with 0.
 
 # What are some examples of information that would be Arrays as opposed to some other data type?
 
-Replace me with your response...
+Obviously arrays are different than strings in that you can put a whole bunch of strings in an array and then they’re indexed by number and you can call then up accordingly, instead of just having a whole load of strings out there individually. Arrays are different than hashes in that arrays are just indexed by number and hashes are indexed by “keys,” but I haven’t gotten that far into hashes yet so I’m still hazy on the significance of this difference. 
 
 # What is one way, using Ruby, to retrieve the 6th element in an Array? How about the 8th element? What happens if you try to retrieve the value of the _9999th_ element (or some element that doesn't exist in the array)?
 
-Replace me with your response...
+Let’s say the array is called this_array. If you want to retrieve the 6th element, you would say “puts this_array[5]” and that will bring back the 6th item in the array since Ruby starts counting at zero. If you want the 8th element just put 7 in the brackets. If you try to retrieve the value of an element that doesn’t exist (like the 9999th element in an array of 200), Ruby answers “nil” meaning there is nothing in that position.
 
 # The previous question asks about finding, for example, the 6th element in an Array. Is it possible to find the **-6th** (Notice the negative symbol!) element in an Array? What does that even mean?
 
-Replace me with your response...
+Yes, you can use the negative positions of an array if you’re looking for a character based on it’s distance from the end of the array. Ruby just counts backward instead of forward. 
 
 # How would you perform an operation on every element inside an Array?
 
-Replace me with your response...
+You do that using the each method. This method will let you do something to each item in an a block of code, but it won’t actually change the original array.
 
 # How do you add elements to an Array?
 
-Replace me with your response...
+You can use the shovel operator (<<), also known as push (same as using “.push”), or simply a “+” sign ([1,2,3] + [4] = [1,2,3,4])  to add something to the end of an array. Unshift lets you add things to the beginning of an array (see below for more on push/unshift).
 
 # Given the Array `["Laura", "Fiona", "Tori"]`, how would you replace `"Fiona"` with `"Florence"` so that you end up with `["Laura", "Florence", "Tori"]`?
 
-Replace me with your response...
+You can replace elements in an array using .replace, but thus far I’m only able to figure out how to replace all of the elements and not just point to a specific one. I’m assuming there is indeed a way to do that. I’m stuck:
```

Comment:
Check http://ruby-doc.org/core-1.9.3/Array.html#method-i-5B-5D-3D and see if that helps. If not, don't hesitate to follow up–I can provide additional guidance.